src/argus/incident/serializers.py

```
@@ -284,11 +284,14 @@ def to_internal_value(self, data: dict):
     def to_representation(self, instance: Event):
         event_repr = super().to_representation(instance)
 
-        type_tuples = [
-            ("value", instance.type),
-            ("display", instance.get_type_display()),
-        ]
-        event_repr["type"] = OrderedDict(type_tuples)
+        try:
+            type_tuples = [
+                ("value", instance.type),
+                ("display", instance.get_type_display()),
+            ]
+            event_repr["type"] = OrderedDict(type_tuples)
+        except:
```

Bare except is always a smell. What problem are you trying to avoid here?

---

In line 467 in `views.py` when trying to access `serializer.data` it otherwise fails, because for some reason due to the `ListSerializer` the validated serializer calls the `to_representation` function, but instead of getting an actual event as instance it gets a dictionary with the data about the event, therefore we get a `KeyError` when trying to access `instance.type`

---

Catch the KeyError and comment why because it sounds like there is a bigger problem here.